elm/web/rhub.py

```diff
@@ -0,0 +1,601 @@
+        ----------
+        soup_inst : obj
+            Active beautiful soup instance for the url associated with a
+            given publication.
+
+        Returns
+        -------
+        category (str)
+        """
+
+        category = soup_inst.find('span',
+                                  {'class': 'type_classification'}).text
+
+        return category
+
+    def _scrape_year(self, soup_inst):
+        """Scrape publication year for a given publication.
+
+        Parameters
+        ----------
+        soup_inst : obj
+            Active beautiful soup instance for the url associated with a
+            given publication.
+
+        Returns
+        -------
+        year (str): the year a record was published
+        """
+        year = soup_inst.find('span', {'class': 'date'}).text
+
+        return year
+
+    def _scrape_id(self, soup_inst):
+        """Scrape the NREL Publication Number for a given publication.
+
+        Parameters
+        ----------
+        soup_inst : obj
+            Active beautiful soup instance for the url associated with a
+            given publication.
+
+        Returns
+        -------
+        NREL Publication Number (str)
+        """
+
+        nrel_id = soup_inst.find('ul', {'class': 'relations keywords'}).text
+
+        return nrel_id
+
+    def build_meta(self):
+        """Build a meta dataframe containing relevant information
+         for publications.
+
+        Returns
+        -------
+        pd.DataFrame
+        """
+        publications_meta = pd.DataFrame(columns=('title', 'nrel_id',
+                                                  'authors', 'year',
+                                                  'url', 'doi',
+                                                  'pdf_url', 'category'))
+        for link in self.all_links[:20]:  # quantity control here #
+            with urlopen(link) as page:
+                html = page.read().decode("utf-8")
+            meta_soup = BeautifulSoup(html, "html.parser")
+
+            title = meta_soup.find('h1').text
+            nrel_id = self._scrape_id(meta_soup)
+            authors = self._scrape_authors(meta_soup)
+            doi = self._scrape_links(meta_soup)[0]
+            pdf_url = self._scrape_links(meta_soup)[1]
+            category = self._scrape_category(meta_soup)
+            year = self._scrape_year(meta_soup)
+
+            new_row = {'title': title,
+                       'nrel_id': nrel_id,
+                       'year': year,
+                       'authors': authors,
+                       'url': link,
+                       'doi': doi,
+                       'pdf_url': pdf_url,
+                       'category': category
+                       }
+
+            publications_meta.loc[len(publications_meta)] = new_row
+
+        return publications_meta
+
+    def download_pdf(self, pdf_dir, txt_dir, soup_inst):
+        """
+        Description
+        ----------
+        Downloads a pdf for a given link
```

Comment:
You don't need the Description header, just have the one-line description on the first line of the docstring immediately following the """